pkg/router/routeapihelpers/validation.go

```diff
@@ -367,3 +367,32 @@ func validateCertificatePEM(certPEM string, options *x509.VerifyOptions) ([]*x50
 
 	return certs, nil
 }
+
+// DeprecatedRouteValidation performs an upgrade validation for
+// a route. This checks for issues that will cause failures in the next
+// OpenShift version.
+func DeprecatedRouteValidation(route *routev1.Route) field.ErrorList {
+	tlsConfig := route.Spec.TLS
+	result := field.ErrorList{}
+
+	if tlsConfig == nil {
+		return result
+	}
+
+	tlsCertFieldPath := field.NewPath("spec").Child("tls").Child("certificate")
+	if len(tlsConfig.Certificate) > 0 {
+		certs, err := cert.ParseCertsPEM([]byte(tlsConfig.Certificate))
+		if err != nil {
+			return result
+		}
+
+		if len(certs) < 1 {
+			return result
+		}
+
+		if certs[0].SignatureAlgorithm == x509.SHA1WithRSA || certs[0].SignatureAlgorithm == x509.ECDSAWithSHA1 {
+			result = append(result, field.Invalid(tlsCertFieldPath, "redacted certificate data", "version 4.16 of OpenShift does not support SHA1"))
```

Comment:
Declare `tlsCertFieldPath` inside the `if` block, closer to use.

Reply:
Done.